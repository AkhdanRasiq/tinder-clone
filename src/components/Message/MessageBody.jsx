import React from 'react'
import Chatbox from './Chatbox'

class MessageBody extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      arrMessages: []
    }
  }

  componentDidUpdate = (prevProps) => {
    if(this.props.a_arrMessages !== prevProps.a_arrMessages)
      this.setState({ arrMessages: this.props.a_arrMessages })
  }

  render() {
    return (
      <div className='messageBodyContainer' id='messageBodyContainer'>
        {this.state.arrMessages.map((msg) => (
          <Chatbox 
            key={msg.timestamp}
            a_messages={msg}
          />
        ))}
      </div>
    )
  }
}

export default MessageBody
