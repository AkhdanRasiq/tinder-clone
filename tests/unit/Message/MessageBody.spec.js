import React from 'react'
import { shallow } from "enzyme"

import MessageBody from "../../../src/components/Message/MessageBody"

const test_a_messages = {
  text      : "Test",
  user      : "Admin",
  timestamp : 1643468925
}

describe("MessageBody.jsx", () => {
  let wrapper

  beforeEach(() => {
    const mockUpObject = {
      focus: () => null,
    };
    global.document.getElementById = jest.fn(() => mockUpObject);
    
    wrapper = shallow(
      <MessageBody
        a_arrMessages={[{
            ...test_a_messages
        }]}
      />
    )
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
