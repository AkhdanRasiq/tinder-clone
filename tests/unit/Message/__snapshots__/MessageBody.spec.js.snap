// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBody.jsx renders properly 1`] = `ShallowWrapper {}`;
